yt_dlp/extractor/tencent.py

```diff
@@ -147,45 +178,53 @@ class VQQVideoIE(VQQBaseIE):
 
     _TESTS = [{
         'url': 'https://v.qq.com/x/page/q326831cny0.html',
-        'md5': '826ef93682df09e3deac4a6e6e8cdb6e',
+        'md5': '84568b3722e15e9cd023b5594558c4a7',
         'info_dict': {
             'id': 'q326831cny0',
             'ext': 'mp4',
             'title': '我是选手：雷霆裂阵，终极时刻',
             'description': 'md5:e7ed70be89244017dac2a835a10aeb1e',
             'thumbnail': r're:^https?://[^?#]+q326831cny0',
+            'format_id': r're:^shd',
+            'has_drm': False,
         },
     }, {
         'url': 'https://v.qq.com/x/page/o3013za7cse.html',
-        'md5': 'b91cbbeada22ef8cc4b06df53e36fa21',
+        'md5': 'cc431c4f9114a55643893c2c8ebf5592',
         'info_dict': {
             'id': 'o3013za7cse',
             'ext': 'mp4',
             'title': '欧阳娜娜VLOG',
             'description': 'md5:29fe847497a98e04a8c3826e499edd2e',
             'thumbnail': r're:^https?://[^?#]+o3013za7cse',
+            'format_id': r're:^shd',
+            'has_drm': False,
         },
     }, {
         'url': 'https://v.qq.com/x/cover/7ce5noezvafma27/a00269ix3l8.html',
-        'md5': '71459c5375c617c265a22f083facce67',
+        'md5': '87968df6238a65d2478f19c25adf850b',
         'info_dict': {
             'id': 'a00269ix3l8',
             'ext': 'mp4',
             'title': '鸡毛飞上天 第01集',
             'description': 'md5:8cae3534327315b3872fbef5e51b5c5b',
             'thumbnail': r're:^https?://[^?#]+7ce5noezvafma27',
             'series': '鸡毛飞上天',
+            'format_id': r're:^shd',
+            'has_drm': False,
         },
     }, {
         'url': 'https://v.qq.com/x/cover/mzc00200p29k31e/s0043cwsgj0.html',
-        'md5': '96b9fd4a189fdd4078c111f21d7ac1bc',
+        'md5': 'fadd10bf88aec3420f06f19ee1d24c5b',
         'info_dict': {
             'id': 's0043cwsgj0',
             'ext': 'mp4',
             'title': '第1集：如何快乐吃糖？',
             'description': 'md5:1d8c3a0b8729ae3827fa5b2d3ebd5213',
             'thumbnail': r're:^https?://[^?#]+s0043cwsgj0',
             'series': '青年理工工作者生活研究所',
+            'format_id': r're:^shd',
```

Comment:
This doesn't need to be regex, but otherwise it looks good to me

Reply:
If use `shd-3`, there is no guarantee that there will be 4 available links in the future.

Maybe there is a better way?

Reply:
You're right, my bad. I was running the wrong tests.